Fixed procedure for Razberry Z-Wave on newest HA OS

jorgror opened the pull request:
## Proposed change
The current documentation seems outdated. I changed it to the steps I used while helping a friend set up his Raspberry Pi 3 with Razberry Z-Wave hat with newest Home-Assistant OS.



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [x] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

jorgror committed:
Fixed procedure for Razberry Z-Wave on newest HA OS 

The current documentation seems outdated. I changed it to the steps I used while helping a friend set up his Raspberry Pi 3 with Razberry Z-Wave hat with newest Home-Assistant OS.

homeassistant commented:
Hi @jorgror, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant.io#16293).

Once you do that we will be able to review and accept this pull request.

Thanks!

jorgror committed:
Corrected spelling of Home Assistant

frenck commented on source/_docs/z-wave/device-specific.markdown:
```diff
@@ -50,6 +50,8 @@ echo -e -n "...turn on/off string from examples above..." | cu -l /dev/zstick -s
 
 ### Razberry Board
 
+GPIO is only supported on 32-bit version of Home Assistant OS.
```
This is no longer true.

jorgror replied:
I based this comment on https://www.home-assistant.io/hassio/installation/
I thought it was relevant to add the info here as this is a GPIO board. If this is no longer the case we should update the documentation in both places.

jorgror replied:
I have reverted this.

frenck commented on source/_docs/z-wave/device-specific.markdown:
````diff
@@ -68,37 +70,10 @@ Reboot your Pi 4 without the Razberry Z-Wave hat first. Then shutdown, add the h
-dtoverlay=pi3-disable-bt
-```
-
-Then disable the Bluetooth modem service:
-
-```bash
-sudo systemctl disable hciuart
-```
-
-Once Bluetooth is off, enable the serial interface via the `raspi-config` tool. After reboot run:
-
-```bash
-sudo systemctl mask serial-getty@ttyAMA0.service
-```
-
-so that your serial interface looks like:
-
-```text
-crw-rw---- 1 root dialout 204, 64 Sep  2 14:38 /dev/ttyAMA0
-```
-at this point simply add your user (homeassistant) to the dialout group:
-
````
This might still be correct, depending on the installation method one is using.

jorgror replied:
I think having all of these commands (which are very hard to actually do on Home Assistant OS) is more confusing than helpful. I struggled alot to figure out what to actually do to enable support in Home Assistant OS as there is so much outdated, wrong and unnescesarry info around and the corret steps for Home Assistant OS was nowhere to be found.

frenck replied:
Yes, I agree with you.
However just deleting things, just because they don't apply to you, isn't helping others that do need this.

Hence, in my above comment, it might be wise to add different sections for specific installation types.

jorgror replied:
I agree that I might have hit that delete button a bit hard. I have added the steps I know works for Home Assistant OS, and I have tried to the best of my abilites looked into what should be needed to enable in a more general matter. I would still like to delete most of these commands as they seems outdated and would probalby cause more harm than good in this document. I base this on that they were commited in 2018 ( https://github.com/home-assistant/home-assistant.io/pull/6162 ) and actually seems to target Home Assistant OS (user homeassistant), and I know throught my experience that these commands are no longer needed on Home Assistant OS.

frenck commented on source/_docs/z-wave/device-specific.markdown:
```diff
@@ -68,37 +70,10 @@ Reboot your Pi 4 without the Razberry Z-Wave hat first. Then shutdown, add the h
 
-<div class='note'>
-
-  If you've installed the Z-Way software, you'll need to ensure you disable it before you install Home Assistant or you won't be able to access the board. Do this with `sudo /etc/init.d/z-way-server stop; sudo update-rc.d z-way-server disable`.
```
This isn't the case anymore?

jorgror replied:
I agree that this should stay in the document.

jorgror committed:
Removed comment about GPIO pins 

This is no longer true

jorgror committed:
Added more documentation on other OS.
Added note about README for overlays. That document is very useful.
Reverted back note about Z-Way software.

homeassistant commented:
Hello @jorgror,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`028b44df51f52e63e50aa8a5b996fb57931bdb1d`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant.io&#x2F;pull&#x2F;16293&#x2F;commits&#x2F;028b44df51f52e63e50aa8a5b996fb57931bdb1d) - This commit has something that looks like an email address (Jorgen.Rorvik@nov.com). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

jorgror committed:
Style fixes

frenck commented on source/_docs/z-wave/device-specific.markdown:
```diff
@@ -67,32 +67,33 @@ Reboot your Pi 4 without the Razberry Z-Wave hat first. Then shutdown, add the h
 
 #### Raspberry Pi 3 procedure
 
+Add the following paramaters to the bottom of the `/boot/config.txt` file.
```
```suggestion
Add the following parameters to the bottom of the `/boot/config.txt` file.

```

frenck commented on source/_docs/z-wave/device-specific.markdown:
````diff
@@ -67,32 +67,33 @@ Reboot your Pi 4 without the Razberry Z-Wave hat first. Then shutdown, add the h
-dtoverlay=pi3-disable-bt
+dtoverlay=disable-bt
 ```
+Reboot your Pi 3. 
````
```suggestion

Reboot your Pi 3.
```

frenck commented on source/_docs/z-wave/device-specific.markdown:
````diff
@@ -67,32 +67,33 @@ Reboot your Pi 4 without the Razberry Z-Wave hat first. Then shutdown, add the h
 ```
 
-Once Bluetooth is off, enable the serial interface via the `raspi-config` tool. After reboot run:
+You should also check the README for details on the overlays. You might find it in `/boot/overlays/README` on your SD-card. If it is not there you can find the official version here: https://github.com/raspberrypi/firmware/blob/master/boot/overlays/README.
````
```suggestion
You should also check the README for details on the overlays. You might find it in `/boot/overlays/README` on your SD-card. If it is not there you can find [the official version here](https://github.com/raspberrypi/firmware/blob/master/boot/overlays/README).
```

frenck commented on source/_docs/z-wave/device-specific.markdown:
```diff
@@ -67,32 +67,33 @@ Reboot your Pi 4 without the Razberry Z-Wave hat first. Then shutdown, add the h
+Reboot your Pi 3. 
 
-Then disable the Bluetooth modem service:
+For Home Assistant OS this should be everything you need to do. You should now be able to use Razberry Z-Wave from ` /dev/ttyAMA0 `
```
```suggestion
For Home Assistant OS this should be everything you need to do. You should now be able to use Razberry Z-Wave from `/dev/ttyAMA0`.
```

frenck commented on source/_docs/z-wave/device-specific.markdown:
```diff
@@ -67,32 +67,33 @@ Reboot your Pi 4 without the Razberry Z-Wave hat first. Then shutdown, add the h
+<div class='note'>
 
-so that your serial interface looks like:
+  It is possible to keep a limited Bluetooth functionality while using Razberry Z-Wave. Check `boot/overlays/README` on miniuart-bt.
```
```suggestion
  It is possible to keep a limited Bluetooth functionality while using Razberry Z-Wave. Check `boot/overlays/README` on `miniuart-bt`.
```

frenck commented on source/_docs/z-wave/device-specific.markdown:
````diff
@@ -67,32 +67,33 @@ Reboot your Pi 4 without the Razberry Z-Wave hat first. Then shutdown, add the h
-```
+<div class='note'>
+
+  disable-bt was previously known as `pi3-disable-bt`. If your OS is old, you might need to use this instead.
````
```suggestion
  `disable-bt` was previously known as `pi3-disable-bt`. If your OS is old, you might need to use this instead.
```

jorgror committed:
Apply suggestions from code review

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

frenck reviewed (approved):
Thanks, @jorgror 👍
